### PR TITLE
ruby-zoom: init at 4.1.0

### PR DESCRIPTION
**Pull request by vmandela**
###### Motivation for this change

This tool is very useful in quickly searching in a git repository and opening the results in an editor. 
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [ ] NixOS
  - [ ] OS X
  - [x] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

From the website,

https://gitlab.com/mjwhitta/zoom

Zoom adds some convenience to ag/ack/grep by allowing you to quickly
open your search results in your editor of choice.

Also available as a ruby gem.

https://rubygems.org/gems/ruby-zoom

### CODE REVIEW

**Comment by Mic92** [comment]
You have to add ncurses as a build input:

```nix
buildInputs = [ ncurses ];
```

File: pkgs/tools/text/ruby-zoom/default.nix [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  inherit ruby;
+  gemfile = ./Gemfile;
+  lockfile = ./Gemfile.lock;
+  gemset = ./gemset.nix;
+
+  meta = with lib; {
+    description = "Quickly open CLI search results in your favorite editor!";
+    homepage    = https://gitlab.com/mjwhitta/zoom;
+    license     = with licenses; gpl3;
+    platforms   = platforms.unix;
```
Comment (Mic92) [inline pkgs/tools/text/ruby-zoom/default.nix]: Do you want to maintain this package?

Comment (vmandela) [inline pkgs/tools/text/ruby-zoom/default.nix]: Yes. I have added myself as the maintainer for this package.

I have already tried adding ncurses as the build input but it gives me the same failure.  Please see the second commit below. I hope I am doing it correctly. 

Do I need to modify gem sources to point to the installed location of ncurses?

Comment (Mic92) [inline pkgs/tools/text/ruby-zoom/default.nix]: This is how it usually works. I do not how passing dependencies to rubygems works in nixos.

**Comment by Mic92** [comment]
@zimbatm do you know, why ncurses is not picked up by the gem, if it is passed as `buildInput` to `bundlerEnv`

**Comment by zimbatm** [comment]
@Mic92 it probably needs to be passed to the `ruby-terminfo` gem derivation. One way to do that is to create a global override for that gem over here: https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/ruby-modules/gem-config/default.nix .

**Comment by Mic92** [comment]
@zimbatm was not aware of this file, @vmandela this what you have to do instead.

**Comment by zimbatm** [comment]
It's also possible to pass your own gemConfig to the bundlerEnv but in that case I think it's useful to have it for all bundlerEnv users.

**Comment by vmandela** [comment]
@zimbatm, thank you for pointing out the solution.

I have modified the global gem-config and I am able to build and run the tool successfully. I have updated the pull request with the working commit.

**Comment by vmandela** [comment]
@Mic92 , @zimbatm , Please let me know if there are any changes required for this PR to be merged.

File: pkgs/tools/text/ruby-zoom/default.nix [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{ lib, bundlerEnv, ruby, stdenv }:
+
+bundlerEnv {
+  name = "ruby-zoom-4.1.0";
```
Comment (vmandela) [inline pkgs/tools/text/ruby-zoom/default.nix]: I have made this change and tested it.

**Review by zimbatm — APPROVED** [review]

**Comment by zimbatm** [comment]
Thanks!

**Comment by vmandela** [comment]
Thank you.

**Pull request merged**